### DISCUSSION TITLE
Polarfire

### DISCUSSION THREAD

**Discussion by liamdiprose** [YosysHQ/yosys#4110] [Q&A] [answered]
Hey everyone,

I really like the BeagleV-Fire SBC, which has a Microchip Polarfire FPGA onboard. However, based of what I've read, the synthesis tools you're supposed to use for it are awful. I wish there was a free alternative.

So, I'm curious what the process of getting yosys' support for the Polarfire involves?

Also, if you know of any supported boards similar to the BeagleV-Fire, please share them :smiley: 

Cheers,
Liam

**Comment by nakengelhardt** [comment, marked as answer]
IIRC the Polarfire is in the same family as the SmartFusion2 so it would probably involve updating `synth_sf2`. Unfortunately SF2 support is a stub so there's quite a bit of work left to do: first updating all the primitives and making sure the basic LUT and FF mapping works, then same for the I/O buffer handling, and then finally adding the memory and dsp inference passes.

And of course that only gets you synthesis, so you still have to go back into the vendor tool for place and route - or add nextpnr support, which is at least twice as much work as the yosys side...

**Comment by briancolgan** [comment]
Support has been added now!
https://github.com/YosysHQ/yosys/pull/4474
